nva-datamodel-java/src/test/java/no/unit/nva/PublicationTest.java

```diff
@@ -277,10 +277,47 @@ private static Publication publicationWithoutEntityDescription() {
         return publication;
     }
 
-    private static Publication publicationWithNoAssociatedArtifacts() {
+    private static Publication publicationWithNoAssociatedArtifactsOrOriginalDoi() {
         var publication = randomPublication();
         publication.setStatus(DRAFT);
         publication.setAssociatedArtifacts(null);
+        publication.getEntityDescription().getReference().setDoi(null);
+        return publication;
+    }
+
+
+    private static Publication publicationWithAdministrativeAgreementAndLink() {
+        var administrativeAgreement = AdministrativeAgreementGenerator.random();
+        var link = AssociatedLinkGenerator.random();
+        return publicationWithAssociatedArtifact(new AssociatedArtifactList(List.of(administrativeAgreement, link)));
+    }
+
+    private static Publication publicationWithAdministrativeAgreementAndUnpublishedFile() {
+        var unpublishedFile = UnpublishedFileGenerator.random();
+        var administrativeAgreement = AdministrativeAgreementGenerator.random();
+        return publicationWithAssociatedArtifact(new AssociatedArtifactList(List.of(administrativeAgreement,
+                unpublishedFile)));
+    }
+
+    private static Publication publicationWithAdministrativeAgreementAndPublishedFile() {
+        var publishedFile = PublishedFileGenerator.random();
+        var administrativeAgreement = AdministrativeAgreementGenerator.random();
+        return publicationWithAssociatedArtifact(new AssociatedArtifactList(List.of(administrativeAgreement,
+                publishedFile)));
+    }
+
+    private static Publication publicationWithOriginalDoi() {
+        var publication = randomPublication();
```

Comment:
randomPublication has a original DOI?

Reply:
Yes, all fields are filled in a randomPublication